### PR TITLE
HOCS-5161: better active/deleted document handling

### PR DESCRIPTION
**Pull request by nickmcmahon01**
This allows the camel queues to update deleted documents, for scenarios where a document is deleted while it is still processing.

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/document/DocumentDataResource.java [hunk 1]
```diff
@@ -105,10 +104,10 @@ private static ResponseEntity<ByteArrayResource> generateFileResponseEntity(S3Do
                 .body(resource);
     }
 
+    @Deprecated(forRemoval = true)
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/document/DocumentDataResource.java:107]: Usages have already been changed? Might as well remove and major bump it? Unless you want to wait a week or so?

File: src/main/java/uk/gov/digital/ho/hocs/document/DocumentDataService.java [hunk 2]
```diff
@@ -94,17 +93,20 @@ public DocumentData getDocumentData(UUID documentUUID) {
...
 
-    public Set<DocumentData> getDocumentsByReferenceAndActionDataUuid(
-            UUID externalReferenceUUID, UUID actionDataUuid, String type) {
-        return documentRepository
-                .findAllByExternalReferenceUUIDAndActionDataItemUuidAndType(externalReferenceUUID, actionDataUuid, type);
+    public Set<DocumentData> getDocumentsByReferenceAndActionDataUuid(UUID externalReferenceUUID, UUID actionDataUuid, String type) {
+        Set<DocumentData> documentData = documentRepository.findAllActiveByExternalReferenceUUIDAndActionDataItemUuidAndType(externalReferenceUUID, actionDataUuid);
+        if(type != null) {
+            return documentData.stream().filter(it -> it.getType().equals(type)).collect(Collectors.toSet());
+        } else {
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/document/DocumentDataService.java:107]: `else` 👋🏽

File: src/main/java/uk/gov/digital/ho/hocs/document/DocumentDataService.java [hunk 3]
```diff
@@ -180,4 +182,14 @@ public void copyDocuments(CopyDocumentsRequest request) {
         log.info("Successfully copied Documents from case: {}, to case: {}",
                 request.getFromReferenceUUID(), request.getToReferenceUUID());
     }
+
+    // Used internally only
+    private DocumentData getDocumentData(UUID documentUUID) {
+        DocumentData documentData = documentRepository.findByUuid(documentUUID);
+        if (documentData != null) {
+            return documentData;
+        } else {
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/document/DocumentDataService.java:191]: `else` 👋🏽

File: src/main/java/uk/gov/digital/ho/hocs/document/DocumentDataResource.java (hunk 1 above)
Comment (nickmcmahon01) [reply src/main/java/uk/gov/digital/ho/hocs/document/DocumentDataResource.java:107]: I've got a separate PR for the removal, maybe merge it in with the EntityActionUUID change?

File: src/main/java/uk/gov/digital/ho/hocs/document/DocumentDataService.java [hunk 4]
```diff
@@ -180,4 +182,14 @@ public void copyDocuments(CopyDocumentsRequest request) {
         log.info("Successfully copied Documents from case: {}, to case: {}",
                 request.getFromReferenceUUID(), request.getToReferenceUUID());
     }
+
+    // Used internally only
+    private DocumentData getDocumentData(UUID documentUUID) {
```
[the remark this answers is not in the document]
Comment (nickmcmahon01) [reply src/main/java/uk/gov/digital/ho/hocs/document/DocumentDataService.java:187]: I mean between camel/service, happy to remove the comment.

**Pull request merged**